test/cpp/util/channelz_sampler.cc

```diff
@@ -0,0 +1,534 @@
+                << " data:" << std::endl;
+      ::google::protobuf::TextFormat::PrintToString(_channel.data(), &data_str);
+      printf("%s", data_str.c_str());
+    }
+    for (const auto& _subchannel : all_subchannels_) {
+      std::cout << "subchannel " << GetSubchannelID(_subchannel)
+                << " data:" << std::endl;
+      ::google::protobuf::TextFormat::PrintToString(_subchannel.data(),
+                                                    &data_str);
+      printf("%s", data_str.c_str());
+    }
+    for (const auto& _server : all_servers_) {
+      std::cout << "server " << GetServerID(_server) << " data:" << std::endl;
+      ::google::protobuf::TextFormat::PrintToString(_server.data(), &data_str);
+      printf("%s", data_str.c_str());
+    }
+    for (const auto& _socket : all_sockets_) {
+      std::cout << "socket " << GetSocketID(_socket) << " data:" << std::endl;
+      ::google::protobuf::TextFormat::PrintToString(_socket.data(), &data_str);
+      printf("%s", data_str.c_str());
+    }
+  }
+
+  // Store a channel in Json
+  void StoreChannelInJson(const grpc::channelz::v1::Channel& channel) {
+    std::string id = grpc::to_string(GetChannelID(channel));
+    std::string type = "Channel";
+    std::string description;
+    ::google::protobuf::TextFormat::PrintToString(channel.data(), &description);
+    StoreEntityInJson(id, type, description);
+  }
+
+  // Store a subchannel in Json
+  void StoreSubchannelInJson(const grpc::channelz::v1::Subchannel& subchannel) {
+    std::string id = grpc::to_string(GetSubchannelID(subchannel));
+    std::string type = "Subchannel";
+    std::string description;
+    ::google::protobuf::TextFormat::PrintToString(subchannel.data(),
+                                                  &description);
+    StoreEntityInJson(id, type, description);
+  }
+
+  // Store a server in Json
+  void StoreServerInJson(const grpc::channelz::v1::Server& server) {
+    std::string id = grpc::to_string(GetServerID(server));
+    std::string type = "Server";
+    std::string description;
+    ::google::protobuf::TextFormat::PrintToString(server.data(), &description);
+    StoreEntityInJson(id, type, description);
+  }
+
+  // Store a socket in Json
+  void StoreSocketInJson(const grpc::channelz::v1::Socket& socket) {
+    std::string id = grpc::to_string(GetSocketID(socket));
+    std::string type = "Socket";
+    std::string description;
+    ::google::protobuf::TextFormat::PrintToString(socket.data(), &description);
+    StoreEntityInJson(id, type, description);
+  }
+
+  // Store an entity in Json
+  void StoreEntityInJson(std::string& id, std::string& type,
+                         std::string& description) {
+    std::string start, finish;
+    gpr_timespec ago = gpr_time_sub(
+        now_,
+        gpr_time_from_seconds(FLAGS_sampling_interval_seconds, GPR_TIMESPAN));
+    std::stringstream ss;
+    time_t tmp = GprTimespec2Timet(now_);
+    ss << std::put_time(std::localtime(&tmp), "%F %T");
```

Comment:
nit: related to above comment about deleting `GprTimespec2Timet`, we can use something like std::localtime(now_.tv_sec) more directly.